docs/Datagrid.md

```diff
@@ -0,0 +1,737 @@
+    const [updateMany, { loading }] = useUpdateMany(
+        'posts',
+        selectedIds,
+        { views: 0 },
+        {
+            onSuccess: () => {
+                refresh();
+                notify('Posts updated');
+                unselectAll('posts');
+            },
+            onFailure: error => notify('Error: posts not updated', { type: 'warning' }),
+        }
+    );
+    const handleClick = () => setOpen(true);
+    const handleDialogClose = () => setOpen(false);
+
+    const handleConfirm = () => {
+        updateMany();
+        setOpen(false);
+    };
+
+    return (
+        <>
+            <Button label="Reset Views" onClick={handleClick} />
+            <Confirm
+                isOpen={open}
+                loading={loading}
+                title="Update View Count"
+                content="Are you sure you want to reset the views for these items?"
+                onConfirm={handleConfirm}
+                onClose={handleDialogClose}
+            />
+        </>
+    );
+}
+
+export default CustomResetViewsButton;
+```
+
+**Tip**: `<Confirm>` leverages material-ui's `<Dialog>` component to implement a confirmation popup. Feel free to use it in your admins!
+
+**Tip**: `<Confirm>` text props such as `title` and `content` are translatable. You can pass translation keys in these props. Note: `content` is only translatable when value is `string`, otherwise it renders the content as a `ReactNode`.
+
+**Tip**: You can customize the text of the two `<Confirm>` component buttons using the `cancel` and `confirm` props which accept translation keys. You can customize the icons by setting the `ConfirmIcon` and `CancelIcon` props, which accept a SvgIcon type.
+
+**Tip**: React-admin doesn't use the `<Confirm>` component internally, because deletes and updates are applied locally immediately, then dispatched to the server after a few seconds, unless the user chooses to undo the modification. That's what we call optimistic rendering. You can do the same for the `<ResetViewsButton>` by setting `undoable: true` in the last argument of `useUpdateMany()`, as follows:
+
+```diff
+// in ./CustomResetViewsButton.js
+import * as React from "react";
+import {
+    Button,
+    Confirm,
+    useListContext,
+    useUpdateMany,
+    useRefresh,
+    useNotify,
+    useUnselectAll,
+} from 'react-admin';
+import { VisibilityOff } from '@material-ui/icons';
+
+const CustomResetViewsButton = () => {
+    const { selectedIds } = useListContext();
+    const refresh = useRefresh();
+    const notify = useNotify();
+    const unselectAll = useUnselectAll();
+    const [updateMany, { loading }] = useUpdateMany(
+        'posts',
+        selectedIds,
+        { views: 0 },
+        {
+            onSuccess: () => {
+                refresh();
+-               notify('Posts updated');
++               notify('Posts updated', { undoable: true }); // the last argument forces the display of 'undo' in the notification
```

Comment:
```suggestion
+               notify('Posts updated', { type: 'info', undoable: true }); // the last argument forces the display of 'undo' in the notification
```

Reply:
Do we really have to ? I think info is the default. If it's not, it should be!

Reply:
That's the default in the redux action creator

Reply:
The problem is that TypeScript complains about it:
`TS2345: Argument of type '{ undoable: true; }' is not assignable to parameter of type 'NotificationType | (NotificationOptions & { type: NotificationType; })'. ...`

Maybe I can change the signature of the `type` to make the type part optional and assign `info` as default value: 

- `NotificationType | (NotificationOptions & { type?: NotificationType })`
- `const { type: messageType = 'info', ...options } = type || {};`